### PR TITLE
feat: wrap starknet transaction

### PR DESCRIPTION
**Pull request by greged93**
Adds a type that wraps a starknet-rs transaction in order to implement `to_eth_transaction`. Also uses `to_eth_transaction` when filtering a starknet transaction.

Time spent on this PR: 1 day

# Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [x] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Testing
- [ ] Other (please describe):

# What is the current behavior?
The current implementation of the codebase generates a lot of boiler plate code, which can be difficult to follow. 

Resolves: #150 and #152 

# What is the new behavior?
Adds this wrapper type around the starknet-rs transaction which allows us to remove a lot of this code in a cleaner manner.

- Adds wrapper type around starknet-rs transaction
- Implement `to_eth_transaction` for this type
- Factor code to use `to_eth_transaction`

# Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: crates/core/src/models/mod.rs [hunk 1]
```diff
@@ -247,3 +264,139 @@ impl ConvertibleStarknetBlock for BlockWithTxs {
...
+}
+
+impl From<Felt252Wrapper> for H256 {
+    fn from(felt: Felt252Wrapper) -> Self {
+        let felt: FieldElement = felt.into();
+        H256::from_slice(&felt.to_bytes_be())
+    }
+}
+
+impl From<Felt252Wrapper> for U256 {
```
Comment (Eikix) [inline crates/core/src/models/mod.rs:289]: i like

File: crates/core/src/models/mod.rs [hunk 2]
```diff
@@ -24,6 +33,14 @@ pub struct TokenBalances {
     pub token_balances: Vec<TokenBalance>,
 }
 
+#[derive(Debug, Error)]
+pub enum ConversionError {
+    #[error("transaction conversion error: {0}")]
+    TransactionConvertionError(String),
```
Comment (ClementWalter) [inline crates/core/src/models/mod.rs:39]: typo "Conver**t**ion"

Comment (greged93) [reply crates/core/src/models/mod.rs:39]: I think it's also conversion in english no?

Comment (ClementWalter) [reply crates/core/src/models/mod.rs:39]: this => `TransactionConvertionError` should be `TransactionConversionError`

**Review by Eikix — APPROVED** [review]
lgtm

**Pull request merged**